apps/mobile/src/components/ChatDetailScreen/components/VexlbotMessageItem/components/TradeChecklistMessageWrapper.tsx

```diff
@@ -0,0 +1,28 @@
+import {useMolecule} from 'bunshi/dist/react'
+import {useAtomValue, useSetAtom} from 'jotai'
+import React, {useEffect} from 'react'
+import {Stack} from 'tamagui'
+import * as fromChatAtoms from '../../../../../state/tradeChecklist/atoms/fromChatAtoms'
+import {chatMolecule} from '../../../atoms'
+
+interface Props {
+  children: React.ReactNode
+}
+
+function TradeChecklistMessageWrapper({children}: Props): JSX.Element {
+  const {chatIdAtom, publicKeyPemBase64Atom} = useMolecule(chatMolecule)
+
+  const chatId = useAtomValue(chatIdAtom)
+  const inboxKey = useAtomValue(publicKeyPemBase64Atom)
+  const setParentChat = useSetAtom(fromChatAtoms.setParentChatActionAtom)
+
+  // sets parent chat for checklist as user can interact with it directly from chat
+  // without going through trade checklist flow
+  useEffect(() => {
+    setParentChat({chatId, inboxKey})
```

Comment:
Shouldn't this be done once for the chat as a whole? It will be called many times unnecessarily for every message instead of once when the chat is rendered...

Reply:
Previously I was also thinking about that, but then I wanted to split the checklist functionality from chat, so chat would not set the checklist state, if you know what I mean. Thats why I did it this way.

Reply:
But as I am thinking about it, you are right, as the checklist is part of chat schema, it is either way connected with it. Will rework it, good point.